lib/limiter/limiter_test.go

```diff
@@ -459,3 +459,44 @@ func mustServeAndReceiveStatusCode(t *testing.T, handler http.Handler, wantStatu
 
 	require.Equal(t, wantStatusCode, response.StatusCode)
 }
+
+func TestRateLimiter_IsRateLimited(t *testing.T) {
+	t.Parallel()
+
+	clock := clockwork.NewFakeClock()
+	limiter, err := NewRateLimiter(Config{
+		Clock: clock,
+		Rates: []Rate{
+			{
+				Period:  time.Minute,
+				Average: 10,
+				Burst:   10,
+			},
+		},
+	})
+	require.NoError(t, err)
+	require.False(t, limiter.IsRateLimited("token1"))
+
+	// Consume some tokens but not all
+	for range 5 {
+		require.NoError(t, limiter.RegisterRequest("token1", nil))
+	}
+
+	require.False(t, limiter.IsRateLimited("token1"))
+
+	// Consume the rest of the tokens
+	for range 4 {
+		require.NoError(t, limiter.RegisterRequest("token1", nil))
+	}
```

Comment:
Shouldn't we confirm the token is still not limited?

```suggestion
	}
	
	require.False(t, limiter.IsRateLimited("token1"))	
```

Otherwise we can just do `for range 5` and not call `require.NoError(t, limiter.RegisterRequest("token1", nil))` below.